tracecat/tables/importer.py

```diff
@@ -25,38 +29,112 @@ def __init__(
+                            raise TracecatImportError("Invalid JSON format") from e
             return value
-        return convert_value(value, col_type)
+        except (ValueError, TypeError) as e:
+            raise TracecatImportError("Cannot convert value") from e
 
     def map_row(
-        self, csv_row: dict[str, str], column_mapping: dict[str, str]
+        self, row: dict[str, Any], column_mapping: dict[str, str]
     ) -> dict[str, Any]:
-        """Map a single CSV row to table columns with type conversion."""
-        mapped_row: dict[str, Any] = {}
+        """Map a CSV row to table schema using column mapping"""
+        mapped_row = {}
 
         for csv_col, table_col in column_mapping.items():
-            if not table_col or table_col == "skip":
+            if not table_col or table_col == "skip" or csv_col not in row:
                 continue
 
+            # Skip columns that don't exist in the table schema
             if col_info := self.columns.get(table_col):
-                value = csv_row[csv_col]
+                value = row[csv_col]
                 try:
                     mapped_row[table_col] = self.convert_value(value, col_info.type)
-                except TypeError as e:
-                    raise TracecatImportError(
-                        f"Cannot convert value {value!r} in column {table_col!r}"
-                        f" to SQL type {col_info.type}"
-                    ) from e
+                except Exception as e:
+                    raise TracecatImportError("Cannot convert value") from e
 
         return mapped_row
 
     async def process_chunk(
         self, chunk: list[dict[str, Any]], service: TablesService, table: Table
     ) -> None:
-        """Process and insert a chunk of rows."""
-        if chunk:
-            count = await service.batch_insert_rows(table, chunk)
+        """Process chunk of rows using Polars for efficiency"""
+        if not chunk:
+            return
+
+        try:
+            # Convert chunk to Polars DataFrame
+            df = pl.DataFrame(chunk)
+
+            for col_name in df.columns:
+                col_info = self.columns.get(col_name)
+                if col_info and col_name in df.columns:
+                    pl_type = self.type_mapping.get(col_info.type)
+
+                    # Special handling for JSONB columns
+                    if col_info.type == SqlType.JSONB:
+                        try:
+                            # Ensure JSON values are proper strings
+                            df = df.with_columns(
+                                pl.col(col_name).map_elements(
+                                    lambda x: json.dumps(x)
+                                    if not isinstance(x, str)
+                                    else x,
+                                    return_dtype=pl.Utf8,
+                                )
+                            )
+                        except Exception as e:
```

Comment:
why do you need special treatment of str? you can serialize json as string too. Also why not polars native APIs?
```
df = df.with_columns(
    pl.col(col_name)
      .struct.json_encode()        # native; no Python call‑backs
      .alias(col_name)
)
```